### DISCUSSION TITLE
Wayland DPMS

### DISCUSSION THREAD

**Discussion by 5ila5** [qtile/qtile#4550] [Q&A]
To turn my monitor off on inactivity i used DPMS settings or `xset dpms force off`. It took me some time to archive something similar on Wayland using [swayidle](https://github.com/swaywm/swayidle) for inactivity detection and a small rust program called [dpms-off](https://github.com/lilydjwg/dpms-off) to turn off the monitor. 
My Question: Is there a better way of doing this (If so are there plans to properly document this or did I just missed the proper documentation) or are there any plans of a better native implementation of at least the dpms-off part?

**Comment by m-col** [comment]
You've pretty much found the correct solution, and so...

> Is there a better way of doing this 

Nope, this is the Correct Way

> are there plans to properly document this or did I just missed the proper documentation

There are no docs or plans for docs for this kind of thing in Qtile's docs, as this is rather a general Wayland world thing. That said, I think a short paragraph mentioning things like this but linking to a more comprehensive, general Wayland documentation could be beneficial. Or maybe even just https://arewewaylandyet.com/. Thoughts?

The main responsibility Qtile has here is supporting the general protocols (in this case [zwlr_output_power_manager_v1](https://wayland.app/protocols/wlr-output-power-management-unstable-v1)).

**Reply by 5ila5** [reply]
I think a general Wayland paragraph is a good idea.

For this problem: its not that easy finding information about this as sway and other compositors implemented this functionality themselfs: `swaymsg output DP-1 dpms off`,` hyprctl dispatcher dpms off`


For anyone searching the same thing I ended up using `swayidle` and `wlopm`:

`swayidle -w timeout 120 'wlopm --off "*"' resume 'wlopm --on "*"'`

**Reply by Jeff-WuYo** [reply]
> For anyone searching the same thing I ended up using `swayidle` and `wlopm`:
> 
> `swayidle -w timeout 120 'wlopm --off "*"' resume 'wlopm --on "*"'`

Did you put this in `config.py`, or at `.config/swayidle/config`, and just call `swayidle`?
Another small issue is when I'm playing video, it still suspend my screen. I think I need something like [SwayAudioIdleInhibit](https://github.com/ErikReider/SwayAudioIdleInhibit) to do it, or is there a native way with swayidle?

**Reply by 5ila5** [reply]
I run some shell scripts on startup using subprocess from my qtile config from a startup function

```python
@hook.subscribe.startup_once
def startup():
    # YOUR CODE HERE
```
